corehq/apps/app_manager/models.py

```diff
@@ -343,9 +343,17 @@ def get_action_properties(self, action):
                 yield name, path
 
 
-class UpdateCaseAction(FormAction):
+class SmartCaseUpdate(DocumentSchema):
```

Comment:
Curious what you would think of a name like `ConditionalCaseUpdate`?

Reply:
I'm worried that's too close to update conditions. Although to be honest I've had a thought we might be able make what we're doing here part of update conditions.

Reply:
To clarify I'm talking about `FormActionCondition`s: [link](https://github.com/dimagi/commcare-hq/blob/dc31e895f212aac850c5663b62ea13203c7e21ce/corehq/apps/app_manager/models.py#L291).

Reply:
That's an interesting idea, using/extending `FormActionCondition`. IIRC, those control whether or not the entire action takes place (not whether or not to act on specific questions), and I think we currently only use them for open/close. It **is** conceptually the same kind of functionality.

Reply:
Yeah, we could make `FormActionCondition` hold a list of questions instead of checking just a single one. But I think  that might be too much of an infrastructure change? 

Going off your idea, maybe rename it `ConditionalCasePropertyUpdate` to show it's specific to a case property.

Reply:
I'd guess it would also take some work to make the condition apply to individual questions...in any case, yeah, agreed with sticking to the spec Simon put together for this.

Defer to @stephherbers on final naming.

Reply:
@orangejenny @AddisonDunn I guess choosing the model name after my namesake will not be a popular choice so I'm leaning towards `ConditionalCaseUpdate` or `ConditionalCasePropertyUpdate` (leaning more towards the first) and will make those changes. Also +1 on sticking to the spec for this iteration

Reply:
`StephConditionalCaseUpdate`